library/src/interactor_impl.cxx

```diff
@@ -266,15 +256,30 @@ class interactor_impl::internals
         else if (keySym == "Return")
         {
           ren->ResetCamera();
-          renWin->Render();
         }
         else if (keySym == "Space")
         {
           self->AnimationManager.ToggleAnimation();
-          renWin->Render();
         }
         break;
     }
+
+    if (checkColoring)
+    {
+      // TODO This is not super great, do better ?
+      bool useCell;
+      std::string arrayName;
+      int component;
+      renWithColor->GetColoring(useCell, arrayName, component);
```

Comment:
What about splitting `GetColoring` in 3 functions so you can call one liner:
```cpp
      self->Options.set("cells", renWithColor->GetUseCell());
      self->Options.set("scalars", renWithColor->GetArrayName());
      self->Options.set("component", renWithColor->GetArrayComponent());
```

Reply:
good point